src/protocolsupport/protocol/utils/types/nbt/NBTCompound.java

```diff
@@ -83,4 +89,18 @@ public int hashCode() {
 		return tags.hashCode();
 	}
 
+	@Override
+	public NBTCompound clone() {
+		//TODO: better clone method
```

Comment:
Perhaps @Shevchik can improve this as he designed the nbt system.

Reply:
i think it just needs a deep-copy. ill try to get that updated before the merge